src/optimizer.cpp

```diff
@@ -53,18 +54,40 @@ void Optimizer::getParams()
   getParam(vx_std_, "vx_std", 0.1);
   getParam(vy_std_, "vy_std", 0.1);
   getParam(wz_std_, "wz_std", 0.3);
+
+  getParam(controller_frequency_, "controller_frequency", 10.0);
   getParam(control_sequence_shift_offset_, "control_sequence_shift_offset", 1);
 
   std::string motion_model_name;
   getParam(motion_model_name, "motion_model", std::string("DiffDrive"));
   auto motion_model = magic_enum::enum_cast<MotionModel>(motion_model_name);
+
   if (motion_model.has_value()) {
     setMotionModel(motion_model.value());
   } else {
     RCLCPP_WARN(logger_, "Motion model is unknown, use default/previous");
   }
 }
 
+void Optimizer::setOffset()
+{
+  const double controller_period = 1.0 / controller_frequency_;
+  constexpr double eps = 1e-6;
+
+  if (controller_period > model_dt_) {
+    RCLCPP_WARN(logger_, "Controller period is more then model dt");
```

Comment:
A note but I think its up to you: Are these items worth users being aware of? If not, then I would suggest making these `RCLCPP_DEBUG`'s instead that can be found in debug logs if there is a problem rather than printing for the usual case.

Reply:
Mppi predicting path considering that you will apply each control for dt. I think when you have controller period more then that value it could work unexpectedly

Reply:
Got it, should we throw an exception then so that we fail the task because it could be unstable / work unexpectedly? I figure that's worth straight up rejecting.

Reply:
Can the other warnings/info be debugs? Or are these needed for users to be aware of?

Reply:
I think this could be helpful to just mention if you set model_dt not equal controller period which is ideal

Reply:
Ok